src/librustc/hir/lowering.rs

```diff
@@ -2284,25 +2578,39 @@ impl<'a> LoweringContext<'a> {
                 let value = self.lower_body(None, |this| this.lower_expr(e));
                 hir::ItemConst(self.lower_ty(t, ImplTraitContext::Disallowed), value)
             }
-            ItemKind::Fn(ref decl, unsafety, constness, abi, ref generics, ref body) => {
+            ItemKind::Fn(ref decl, header, ref generics, ref body) => {
                 let fn_def_id = self.resolver.definitions().local_def_id(id);
+
                 self.with_new_scopes(|this| {
+                    // Note: we can use non-async decl here because lower_body
+                    // only cares about the input argument patterns,
+                    // not the return types.
```

Comment:
Not sure what this comment is trying to convey - what is `decl` and where exactly is "here"?

Reply:
Edited the comment to "Note: we don't need to modify the return type from T to impl Future<Output = T> because lower_body doesn't use the return type"